examples/OpenGLWindow/LoadShader.cpp

```diff
@@ -5,86 +5,116 @@
 #include <stdlib.h>
 
 
-// Load the shader from the source text
-void gltLoadShaderSrc(const char *szShaderSrc, GLuint shader)
+// Load the shader from the header / source text. header can be empty/null
+void gltLoadShaderSrc(const char *szShaderHdr, const char *szShaderSrc, GLuint shader)
 {
-	GLchar *fsStringPtr[1];
-
-	fsStringPtr[0] = (GLchar *)szShaderSrc;
-	glShaderSource(shader, 1, (const GLchar **)fsStringPtr, NULL);
+	GLchar *fsStringPtr[2];
+	fsStringPtr[0] = szShaderHdr ? (GLchar *)szShaderHdr : (GLchar *)szShaderSrc;
+	fsStringPtr[1] = (GLchar *)szShaderSrc;
+	glShaderSource(shader, szShaderHdr ? 2 : 1, (const GLchar **)fsStringPtr, NULL);
 }
 
 
-GLuint gltLoadShaderPair(const char *szVertexProg, const char *szFragmentProg)
+// Load the shader, any of the parameters can be empty / null
+GLuint gltLoadShaderTriple(const char *szHeader, const char *szVertexProg, const char *szGeometryProg, const char *szFragmentProg)
```

Comment:
How about keeping the old gltLoadShaderPair, and add a new gltLoadShaderPairTriple? The old code uses gltLoadShaderPair etc?

Reply:
Your'd prefer to avoid changes to gltLoadShaderPair() completely?

It might look like a lot of changes, but changes to make gltLoadShaderTriple()  out of gltLoadShaderPair are trivial.   A bit of reordering and a couple of if(), to allow loading of shaders conditionally.    

And  gltLoadShaderPair() should work - it just calls gltLoadShaderTriple(), with szGeometryProg = 0.